controllers/postgresql_controller.go

```diff
@@ -61,7 +67,7 @@ func (a *postgreSQLAdapter) getServiceCommonSpec() *v1alpha1.ServiceCommonSpec {
 }
 
 func (a *postgreSQLAdapter) getUserConfig() any {
-	return &a.Spec.UserConfig
+	return a.Spec.UserConfig
```

Comment:
Nice catch. Let's make this same way to other types. Is it returns a pointer to the pointer right now (in main)?